supreme_court_predictions/statistics/descriptives.py

```diff
@@ -0,0 +1,246 @@
+                )
+            elif "appellant" in role:
+                clean_roles["for appellant"] = clean_roles.get("appellant") + 1
+            else:
+                clean_roles[role] = clean_roles.get(role, 0) + 1
+
+        return clean_roles
+
+    def get_conversation_desc(self):
+        """
+        [Insert]
+        """
+
+        # Load data
+        conversations = pd.read_csv(self.local_path + "conversations_df.csv")
+
+        # For winning_side:
+        # 1 = for petitioner, 0 = for respondent, 2 = unclear,
+        # -1 = unavailable
+
+        # Get dataframe of winning side descriptive stats
+        conversation_stats = self.get_count_desc(
+            conversations, ["winning_side"]
+        )
+
+        # Add case count descriptives
+        conversation_stats.loc[("cases", ""), :] = len(
+            conversations.loc[:, "case_id"].unique()
+        )
+
+        return conversation_stats
+
+    def get_speaker_desc(self):
+        """
+        [Insert]
+        """
+        pass
+
+        # Load data
+        speakers = pd.read_csv(self.local_path + "speakers_df.csv")
+
+        # Get descriptive stats
+        speakers_stats = self.get_count_desc(
+            speakers, ["speaker_type", "speaker_role"]
+        )
+
+        # Add count of unique speaker names and keys
+        speakers_stats.loc[("speaker_name", ""), :] = len(
+            speakers.loc[:, "speaker_name"].unique()
+        )
+        speakers_stats.loc[("speaker_key", ""), :] = len(
+            speakers.loc[:, "speaker_key"].unique()
+        )
+
+        return speakers_stats
+
+    # TODO- may not need
+    def get_utterances_desc(self):
+        """
+        [Insert]
+        """
+        pass
+
+    def get_voters_desc(self):
+        """
+        [Insert]
+        """
+        # Load data
+        voters = pd.read_csv(self.local_path + "voters_df.csv")
+
+        # Get descriptive stats of votes
+        voter_stats = self.get_count_desc(voters, ["vote"])
+        voter_stats.loc[("justices", ""), :] = len(
+            voters.loc[:, "voter"].unique()
+        )
+
+        return voter_stats
+
+    def parse_all_data(self):
+        """
+        [Insert]
+        """
+
+        print("Grabbing advocates descriptive statistics...")
+        self.advocates_stats = self.get_advocate_desc()
+
+        print("Grabbing conversations descriptive statistics...")
+        self.conversations_stats = self.get_conversation_desc()
+
+        print("Grabbing speakers descriptive statistics...")
+        self.speakers_stats = self.get_speaker_desc()
+
+        print("Grabbing voters descriptive statistics...")
+        self.voters_stats = self.get_voters_desc()
+
+        if self.save_data:
+            desc_out = self.output_path + "/descriptive_statistics.xlsx"
+            self.advocates_stats.to_csv(
+                self.output_path + "/advocates_stats.csv",
+                index=True,
+                encoding=ENCODING_UTF_8_SIG,
```

Comment:
Do you need the `utf-8 sig` value or just `utf-8`?

Reply:
i have no idea, i was copying fede's code @FedericoDM

Reply:
> The Unicode Standard permits the BOM in [UTF-8](https://en.wikipedia.org/wiki/UTF-8),[[3]](https://en.wikipedia.org/wiki/Byte_order_mark#cite_note-3) but does not require or recommend its use.[[4]](https://en.wikipedia.org/wiki/Byte_order_mark#cite_note-4)
- https://en.wikipedia.org/wiki/Byte_order_mark#UTF-8

It looks like we should roll with the other UTF-8 instead. Let me know if I'm missing something, @FedericoDM.